packages/kbn-router-utils/src/get_router_link_props/index.ts

```diff
@@ -8,18 +8,19 @@
 
 export interface RouterLinkProps {
   href: string | undefined;
-  onClick: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  onClick: (event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement, MouseEvent>) => void;
```

Comment:
I added the `HTMLButtonElement` so I can use `getRouterLinkProps` also for button props.